frontend/src/components/SaveTranscript/SaveModal.jsx

```diff
@@ -2,35 +2,40 @@ import React, { useState, useRef, useEffect } from 'react';
 import { db } from '../../../../backend/firebase/firebase.config.js';
 import { doc, updateDoc, arrayUnion, getDoc, setDoc, collection, addDoc } from 'firebase/firestore';
 import { useAuth } from '../auth/auth.jsx';
+import './SaveModal.css'
 
 const SaveModal = ({ isOpen, onClose, onSave }) => {
     const [transcriptName, setTranscriptName] = useState('');
     const folderNameRef = useRef(null);
     const [error, setError] = useState('');
     const [folders, setFolders] = useState([])
-    const [selectedFolder, setSelectedFolder] = useState(folders[0] || '');
     const { currentUser } = useAuth();
+    const [selectedFolder, setSelectedFolder] = useState(folders[0] || `${currentUser.displayName}'s Default Folder`);
+    const [showSuccessMessage, setShowSuccessMessage] = useState(false);
 
 
     if (!isOpen) return null;
 
 
-        const fetchFolders = async () => {
-            const userDocRef = doc(db, "users", currentUser.uid);
-            const userDoc = await getDoc(userDocRef);
+    const fetchFolders = async () => {
```

Comment:
Yeah, for all these interactions with db, let's make sure empty states are being handled (logic works as expected for new users when the db is not yet created).